hopp/simulation/technologies/resource/cambium_data.py

```diff
@@ -0,0 +1,316 @@
+                    url="{base}?project_uuid={project_uuid}&scenario={scenario}&location_type={location_type}&latitude={latitude}&longitude={longitude}&year={year}&time_type={time_type}&metric_col=*_MWh&technology={technology}".format(
+                    base=CAMBIUM_BASE_URL, project_uuid=self.project_uuid, scenario=self.scenario, location_type=self.location_type,
+                    latitude=self.latitude, longitude=self.longitude, year=self.year_to_check, time_type=self.time_type, technology=self.technology_map[metric]
+                    )
+                    r = requests.get(url)
+                    if r:
+                        response_dict[metric] = json.loads(r.text)['message'][0]['values']
+                    elif r.status_code == 400 or r.status_code == 403:
+                        print(r.url)
+                        err = r.text
+                        text_json = json.loads(r.text)
+                        if 'errors' in text_json.keys():
+                            err = text_json['errors']
+                        raise requests.exceptions.HTTPError(err)
+                    elif r.status_code == 404:
+                        print(filename)
+                        raise requests.exceptions.HTTPError
+                    elif r.status_code == 429:
+                        raise RuntimeError("Maximum API request rate exceeded!")
+                    else:
+                        n_tries +=1
+                # Save the response dict as a csv file
+                localfile = open(filename, mode="w+")
+                w = csv.writer(localfile)
+                w.writerow(list(response_dict.keys()))
+                w.writerows(zip(*list(response_dict.values())))
+                localfile.close()
+                if os.path.isfile(filename):
+                    success = True
+                    break
+            except requests.exceptions.Timeout:
+                time.sleep(0.2)
+                n_tries +=1
+
+        return success
+
+    def download_resource(self):
+        success = self.call_api(filename=self.filename)
+
+        return success
+
+    #NOTE: format_data() and data() not used in current implementation, these were originally setup for processing data from single file, LCA requires up to 5 cambium files and saving all data may be too large / memory intensive
+    #NOTE: As an alterative to loading all Cambium data into memory, current logic stores file names in self.resource_files which can be used to load data when needed for LCA calculations
+    def format_data(self):
+        """
+        """
+        if not os.path.isfile(self.filename):
+            raise FileNotFoundError(f"{self.filename} does not exist. Try `download_resource` first.")
+
+        self.data = self.filename
+
+    @Resource.data.setter
+    def data(self, data_dict):
+        pass
```

Comment:
They are part of the parent resource class, I was running into some where errors without them defined because of the attrs used in the parent Resource class. Open to removing them if we can find a clean way to do so.